charmrepo/charmstore_test.go

```diff
@@ -190,6 +190,59 @@ func (s *charmStoreRepoSuite) TestLatest(c *gc.C) {
 	}
 }
 
+func (s *charmStoreRepoSuite) TestResolve(c *gc.C) {
+	// Add some charms to the charm store.
+	s.addCharm(c, "~who/trusty/mysql", "mysql")
+	// Use different charms so that revision is actually increased
+	s.addCharm(c, "~who/precise/wordpress", "logging")
+	s.addCharm(c, "~who/precise/wordpress", "wordpress")
+	s.addCharm(c, "~dalek/utopic/riak", "wordpress")
+	s.addCharm(c, "~dalek/utopic/riak", "riak")
+	s.addCharm(c, "~dalek/utopic/riak", "wordpress")
+	s.addCharm(c, "~dalek/utopic/riak", "riak")
+
+	// Define the tests to be run.
+	tests := []struct {
+		id  string
+		url string
+		err string
+	}{{
+		id:  "~who/mysql",
+		url: "cs:~who/trusty/mysql",
+	}, {
+		id:  "~who/trusty/mysql",
+		url: "cs:~who/trusty/mysql",
+	}, {
+		id:  "~who/wordpress",
+		url: "cs:~who/precise/wordpress",
+	}, {
+		id:  "~who/wordpress-1",
+		url: "cs:~who/precise/wordpress-1",
+	}, {
+		id:  "~dalek/riak",
+		url: "cs:~dalek/utopic/riak",
+	}, {
+		id:  "~dalek/utopic/riak-2",
+		url: "cs:~dalek/utopic/riak-2",
```

Comment:
It holds wordpress content but that content is still associated with the riak reference.